### PR TITLE
⚡  Write a deepgramTranscribeVideo() Function using Deno #4476 

### PR DESCRIPTION
**Pull request by Rinshin-Jalal**
## Description

According to the issue [#4476](https://github.com/appwrite/appwrite/issues/4476) on appwrite,

I created a Python Cloud Function for transcribing video using [Deepgram](https://deepgram.com/)

It can transcribe videos using Deepgram

___

### 1. Success full response 

image

### 2. Error Response - Invalid fileUrl
image

## 3. Error Response - Invalid API key

image

### CODE REVIEW

**Comment by vasucp1207** [comment]
Hey, @Rinshin-Jalal I saw your code as I am also working on the same kind of issue can you please tell me the steps you follow in order to run your deno function? I get some errors while executing the commands.

**Comment by Rinshin-Jalal** [comment]
hi, @vasucp1207 

Can you show me the error??



2. Enter this function folder and build the code:

```
docker run -e INTERNAL_RUNTIME_ENTRYPOINT=src/mod.ts --rm --interactive --tty --volume $PWD:/usr/code openruntimes/deno:v2-1.21 sh /usr/local/src/build.sh
```

As a result, a `code.tar.gz` file will be generated.

3. Start the Open Runtime:

```
docker run -p 3000:3000 -e INTERNAL_RUNTIME_ENTRYPOINT=src/mod.ts -e INTERNAL_RUNTIME_KEY=secret-key --rm --interactive --tty --volume $PWD/code.tar.gz:/tmp/code.tar.gz:ro openruntimes/deno:v2-1.21 sh /usr/local/src/start.sh
```

- Docker must be installed
- your code must be in the src folder of this terminal command

**Comment by vasucp1207** [comment]
Hey, @Rinshin-Jalal even though I try to run the same code as yours and also follow all the steps but still got the same error.


Screenshot 2022-10-20 at 11 48 26 PM

**Comment by Rinshin-Jalal** [comment]
> Hey, @Rinshin-Jalal even though I try to run the same code as yours and also follow all the steps but still got the same error.
> 
> 
> Screenshot 2022-10-20 at 11 48 26 PM
> 

Oops, I don't know the answer to this, can you ask it in the appwrite discord server

**Comment by PineappleIOnic** [comment]
Thank you so much for the PR 🤩. We're adding the `hacktoberfest-accepted` label to ensure this PR counts towards your Hacktoberfest contributions count. With that said, please stay active on this PR to address any comments once you receive a review. Happy Hacktoberfest! 🎃

**Comment by Rinshin-Jalal** [comment]
@stnguyen90 Review this

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Great PR! 🤯 We left some comments during the review, please check them out.

File: deno/deepgram-transcribe-video/src/mod.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+export default async function (req: any, res: any) {
+  const APIkey = req.variables["DEEPGRAM_API_KEY"];
```
Comment (stnguyen90) [inline deno/deepgram-transcribe-video/src/mod.ts:2]: best practice for variable names in deno/typescript is to be camel-cased:
```suggestion
  const apiKey = req.variables["DEEPGRAM_API_KEY"];
```

File: deno/deepgram-transcribe-video/src/mod.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  const fileUrl = JSON.parse(req.payload)["fileUrl"];
+
+  // error handling
+  if (!APIkey) {
+    res.json({
+      success: false,
+      message: "Missing Deepgram API key",
+    });
+  }
```
Comment (stnguyen90) [inline deno/deepgram-transcribe-video/src/mod.ts:11]: you need to make sure to call return or else the rest of the code will still execute.

File: deno/deepgram-transcribe-video/src/mod.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+export default async function (req: any, res: any) {
+  const APIkey = req.variables["DEEPGRAM_API_KEY"];
+  const fileUrl = JSON.parse(req.payload)["fileUrl"];
```
Comment (stnguyen90) [inline deno/deepgram-transcribe-video/src/mod.ts:3]: Make sure to catch errors. Passing an invalid JSON string for payload results in an unhelpful error:

```shell
curl http://localhost:3000/ -d '{"variables": {},"payload": "{\"filUrl\":\"https://static.deepgram.com/examples/interview_speech-analytics.wav\""}' -H "X-Internal-Challenge: secret-key" -H "Content-Type: application/json"
{
  "stdout": "",
  "stderr": "Code file not found."
}
```

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
please see my comments.

**Review by stnguyen90 — APPROVED** [review]
Hey 👋 awesome work on your PR! We've approved your work and it'll be merged soon!

**Comment by christyjacob4** [comment]
THANK YOU! All changes merged 🥳

Please reach out to me on our [Discord server](https://appwrite.io/discord) if you would like to claim your Appwrite swags! As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord.  

You can accept the invite by visiting https://github.com/orgs/appwrite/invitation. By joining our team, you will officially be an Appwrite maintainer on GitHub.

You can change your membership [visibility settings](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership), so your new Appwrite team membership badge will show up on your personal GitHub profile.

Please feel free to look for more PRs you might be interested in helping with on our long list of [Hacktoberfest friendly issues](https://github.com/search?q=org%3Aappwrite+org%3Autopia-php+org%3Aopen-runtimes+label%3Ahacktoberfest+created%3A%3E2021-01-01+state%3Aopen&type=issues&s=updated&o=asc) and help make Appwrite better :)

**Pull request merged**